### PR TITLE
Add methods to get number of policies and templates in a PolicySet

### PR DESCRIPTION
**Pull request by jv-garcia**
## Description of changes

Add methods to get number of policies and templates in a PolicySet.

Regarding unit testing, I decided to just reuse this API in previous places where the size of existing iterators are used, it feels like more effort would feel silly here.

## Issue #, if available
#1179

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [ ] A breaking change requiring a major version bump to `cedar-policy` (e.g., changes to the signature of an existing API).
- [ X] A backwards-compatible change requiring a minor version bump to `cedar-policy` (e.g., addition of a new API).
- [ ] A bug fix or other functionality change requiring a patch to `cedar-policy`.
- [ ] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)
- [ ] A change (breaking or otherwise) that only impacts unreleased or experimental code.

I confirm that this PR (choose one, and delete the other options):

- [ X] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).
- [ ] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [ X] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.
- [ ] Requires updates, and I have made / will make these updates myself. (Please include in your description a timeline or link to the relevant PR in `cedar-spec`, and how you have tested that your updates are correct.)
- [ ] Requires updates, but I do not plan to make them in the near future. (Make sure that your changes are hidden behind a feature flag to mark them as experimental.)
- [ ] I'm not sure how my change impacts `cedar-spec`. (Post your PR anyways, and we'll discuss in the comments.)

### CODE REVIEW

File: cedar-policy/CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Starting with version 3.2.4, changes marked with a star (*) are _language breaki
 
 ## [Unreleased]
 Cedar Language Version: TBD
+* Add convenience methods to see how many policies and templates a policy set has (#1179).
```
Comment (cdisselkoen) [inline cedar-policy/CHANGELOG.md:14]: nit: can we put this in the `Added` section (three lines below)?

File: cedar-policy/src/api.rs [hunk 2]
```diff
@@ -2057,6 +2057,18 @@ impl PolicySet {
         self.ast.is_empty()
     }
 
+    /// Returns the number of `Template`s in the `PolicySet`.
```
Comment (cdisselkoen) [inline cedar-policy/src/api.rs:2060]:
```suggestion
    /// Returns the number of `Policy`s in the `PolicySet`.
```

**Review by cdisselkoen — COMMENTED** [review]
Looks great, two nits

File: cedar-policy/src/api.rs (hunk 2 above)
Comment (jv-garcia) [reply cedar-policy/src/api.rs:2060]: Ah, copy paste mishap, let me fix this.

**Review by cdisselkoen — APPROVED** [review]

**Review by shaobo-he-aws — APPROVED** [review]

**Comment by jv-garcia** [comment]
One of the checks failed, is this a known issue? I'd probably need a bit of help to fix it if needed.

**Comment by cdisselkoen** [comment]
It's a problem unrelated to the change in this PR.  Given that, happy to merge this now -- looks like we have 2 approvals and the CI failures are unrelated.

**Pull request merged**